Fixed an out-of-bounds error when querying validator balance via pubkey

jclapis opened the pull request:
This PR fixes #2427. Before, the validator balance RPC routes didn't do bounds checking on the indices of validators when they were passed in via pubkey instead of via index. This would cause a problem when querying the balance of a validator that is in the pending state, because it doesn't have an index yet. This PR adds bounds checking to fix that problem: if the validator is out-of-bounds because it's pending, this will return its effective balance instead of trying to get its actual balance from `state.balances`.

There is an alterative approach, which is to make these routes behave the way they do when indices are passed in rather than pubkeys: if an index is out of bounds, it simply ignores that validator. Since that isn't the way this was originally set up, I decided not to go down that route and replicated the original behavior prior to my changes in #2408.

jclapis committed:
Validator APIs now return actual balance instead of effective balance

jclapis committed:
Fixed an out-of-bounds error when querying validator balance via pubkey

jclapis committed:
Merge branch 'unstable' of https://github.com/status-im/nimbus-eth2 into actual-balance-fix

cheatfate commented on beacon_chain/rpc/beacon_api.nim:
```diff
@@ -235,10 +235,13 @@ proc installBeaconApiHandlers*(rpcServer: RpcServer, node: BeaconNode) =
                                           current_epoch)
             if includeFlag:
+              var validatorBalance = validator.effective_balance
+              if index < len(state.validators):
```
this requirement `index < len(state.validators)` is always `true`, because we are iterating over `state.validators`.

cheatfate commented on beacon_chain/rpc/beacon_api.nim:
```diff
@@ -263,10 +266,13 @@ proc installBeaconApiHandlers*(rpcServer: RpcServer, node: BeaconNode) =
                                             current_epoch)
               if includeFlag:
+                var validatorBalance = validator.effective_balance
+                if index < len(state.validators):
```
This check is also not needed because it was already made at line 247.

jclapis committed:
Fixed some copy/paste errors

The validator routes now compare the index from the validator to the length of `state.balances` like they're supposed to

jclapis commented:
This isn't actually a fix, see #2427 for more info. Ignore this PR.
